### DISCUSSION TITLE
Visual representation of relationships between data

### DISCUSSION THREAD

**Discussion by kgodey** [mathesar-foundation/mathesar#2287] [Feature Ideas]
We'd like to offer a view that lets users visualize their entire schema or database in one place. We'd probably visualize this as an entity-relationship diagram.

---
*If you'd like us to build this feature, please comment below with more details about how you plan to use this functionality.*

**Comment by acxz** [comment]
I would plan to use this functionality as follows:

When onboarding folks onto the database, having an interactive overview of how everything is connected is critical. It greatly speeds up onboarding and having new users becoming familar with the database.

More importantly, having the schema at hand is quite useful when creating input forms to add data into the database. Often times the input form "schema" is at a different abstraction level, then the schema/tables in the database itself. Similar to the benefit that would be seem for input forms, the reverse operation, querying, would also see a benefit as looking at the ERD to write the appropriate query becomes much smoother.

If possible, there should be a graphical representation that shows how the input fields of a form relate to various tables and how the columns of a view table/exploration relate to various tables in the database.
